cmd/pileup/pileup.go

```diff
@@ -0,0 +1,117 @@
+// Command Group: "SAM Tools"
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/sam"
+	"log"
+	"strings"
+)
+
+func usage() {
+	fmt.Print(
+		"pileup - Count bases from sequencing data\n\n" +
+			"Usage:\n" +
+			"  pileup [options] in.bam\n\n" +
+			"Options:\n")
+	flag.PrintDefaults()
+}
+
+type Settings struct {
+	minDp int
+}
+
+// pileFilters parses the Settings struct to make functions to filter the output piles.
+func pileFilters(s Settings) []func(p sam.Pile) bool {
+	var filters []func(p sam.Pile) bool
+
+	if s.minDp > 0 {
+		filters = append(filters, func(p sam.Pile) bool {
+			var count int
+			for i := range p.Count {
+				count += p.Count[i]
+			}
+			for _, i := range p.InsCount {
+				count += i
+			}
+			return count >= s.minDp
+		})
+	}
+
+	return filters
+}
+
+func pileup(infile string, outfile string, s Settings) {
+	reads, recycle, header := sam.GoReadToChanRecycle(infile, 1000)
+	sendChan := make(chan sam.Sam) // no buffer to satisfy recycle contract
+
+	// goroutine sends from the read channel to an unbuffered channel
+	// then returns the previous struct to the recycle channel.
+	go func(<-chan *sam.Sam, chan<- *sam.Sam, chan<- sam.Sam) {
+		var prevRead *sam.Sam
+		for read := range reads {
+			sendChan <- *read
+			if prevRead != nil {
+				recycle <- prevRead
+			}
+			prevRead = read
+		}
+		close(sendChan)
+	}(reads, recycle, sendChan)
+
+	pileChan := sam.GoPileup(sendChan, header, false, nil, pileFilters(s))
+
+	var err error
+	output := fileio.EasyCreate(outfile)
+	_, err = fmt.Fprintln(output, "Chr\tPos\tA\tC\tG\tT\tN\tDEL\tINS")
```

Comment:
I am thinking this is the header line.  Can we start this with a '#'?  I don't think there is a set standard, but I think a decent amount of file formats follow that convention.  This is also the information I think would be helpful to have in the help message